server/src/main/java/org/elasticsearch/action/resync/TransportResyncReplicationAction.java

```diff
@@ -184,4 +186,22 @@ public void handleException(TransportException exp) {
             });
     }
 
+    /**
+     * A proxy for primary-replica resync operations which are performed on replicas when a new primary is promoted.
+     * Replica shards which fail to execute these resync operation will be failed but won't be marked as stale
+     * (eg. keep in the in sync set). This is a best-effort to avoid marking shards as stale during cluster restart.
```

Comment:
remove the comment "(eg. keep in the in sync set)". Also why is this best-effort now? By failing the shard, we're forcing the shard copies to resync.